app/main.py

```diff
@@ -1,2 +1,91 @@
+from dataclasses import dataclass
+from typing import Hashable, Any
+
+RESIZE_THRESHOLD = 0.7
+INITIAL_CAPACITY = 8
+
+
+@dataclass
+class Node:
+    key: Hashable
+    value: Any
+
+
 class Dictionary:
-    pass
+    def __init__(self, capacity: int = INITIAL_CAPACITY) -> None:
+        self.capacity = capacity
+        self.size = 0
+        self.hash_table: list[Node | None] = [None] * self.capacity
+
+    def _calculate_index(self, key: Hashable) -> int:
+        index = hash(key) % self.capacity
+
+        while (
+            self.hash_table[index] is not None
+            and self.hash_table[index].key != key
+        ):
+            index = (index + 1) % self.capacity
+        return index
+
+    @property
+    def current_max_size(self) -> float:
+        return self.capacity * RESIZE_THRESHOLD
```

Comment:
The `current_max_size` method calculates the threshold for resizing based on `RESIZE_THRESHOLD`. This might not be optimal if the threshold is too low, as it could lead to frequent resizing. Consider reviewing the threshold value to ensure efficient resizing.